### DISCUSSION TITLE
Discord

### DISCUSSION THREAD

**Discussion by The-Futurist** [antlr/antlr4#4036] [Ideas]
Is there any prospect of a Discord channel being setup for Antlr? I rely on these channels sometimes for quick tech chats, like the C# language and stuff, I didn't see one for Antlr and I do not much about setting these up so wanted to raise the idea.

**Comment by kaby76** [comment]
Why Discord? We already have several ways of communicating, don't we?
* https://groups.google.com/g/antlr-discussion -- for "announcements", but seems to be used for Q+A.
* http://stackoverflow.com/questions/tagged/antlr4 -- for Q+A on specific problems with grammar, parser, build environment, etc.
* https://github.com/antlr/antlr4/discussions -- new features
* https://github.com/antlr/antlr4/issues -- bugs and issues (aka "to do") for the repo
* https://github.com/antlr/grammars-v4/issues -- bugs and issues for the repo (grammar specific problems).
* etc.

**Reply by The-Futurist** [reply]
Well fair enough, but I don't use Google Groups, hate stackoverflow (apparently I don't ask "interesting" questions) and yes Github works but is not interactive, like I can visit the C# Discord channel and converse in close to real-time.

**Reply by kaby76** [reply]
I agree with you w.r.t. StackOverflow.com. Generally, questions cannot ask for opinions. Questions and answers are subject to (1) anonymous positive and negative voting, which is pretty useless; (2), editing outside my control. Consequently, I almost never post SO Answers only Comments. But, when I do comment, I try to first check my work: generate a driver, test the input on the parser, make corrections, test, verify, create a repo, wordsmith an answer, check with Grammarly, and post.

**Reply by DrCorchit** [reply]
@kaby76 I think the motivation for creating a discord is as follows:

Each of the collaboration groups you listed above is either unpopular, arcane, or actively hostile to users. Nobody uses google groups. Github is fine, but it's really only for developers who are interested in contributing to the project. And stackoverflow has enormous personality issues that have been discussed in depth many times, but never seem to be resolved.

Discord is nice because it fosters socialization and community. You can chat about unrelated stuff while still maintaining dedicated channels for purely technical conversation.

I do realize that maintaining a discord may be a lot of work, but I strong recommend making antlr discussion more accessible.

**Reply by mike-lischke** [reply]
Discussion channels like Discord tend to receive a huge amount of unrelated jabber, making it really difficult to read a question and its discussion/answers (depending on the overall traffic) unless you can discuss in threads, like Slack allows. People often have a hard time to stay on topic, especially when only one channel exists for all communication.

**Reply by Logopher** [reply]
Discord does have the ability to make "forum-style" channels where conversations are organized into threads.
